pages/client-libraries/graphql.mdx

```diff
@@ -0,0 +1,162 @@
+which includes [Memgraph database](/getting-started/install-memgraph/docker#other-available-docker-images),
+[Memgraph Lab](/data-visualization#quick-start) and [Memgraph MAGE](/advanced-algorithms#quick-start).
+Memgraph uses Bolt protocol to communicate with the client using the exposed
+7687 port. Memgraph Lab is a web application you can use to visualize the data.
+It's accessible at http://localhost:3000 if Memgraph Platform is running
+correctly. The 7444 port enables Memgraph Lab to access and preview the logs,
+which is why both of these ports need to be exposed.
+
+For more information visit the getting started guide on
+[how to run Memgraph with Docker](/getting-started/install-memgraph/docker).
+
+
+### Create a work directory
+
+Next, create a directory for your project and positioning yourself in it:
+
+```
+mkdir graphql-example
+cd graphql-example
+```
+
+### Initialize the Node.js project
+
+Initialize the project using the following code:
+
+```
+npm init es6 --yes
+```
+
+### Setup the Neo4j GraphQL Library
+
+Install the library and its dependencies by running the following command:
+
+```
+npm install @neo4j/graphql graphql neo4j-driver @apollo/server
+```
+
+### Create the application file
+
+The application file will be used to setup the Neo4j GraphQL Library and connect to the running Memgraph instance.
+
+Create the file and add the content below.
+
+```
+touch index.js
+```
+
+```js
+const { Neo4jGraphQL } = require("@neo4j/graphql");
+const { ApolloServer, gql } = require("apollo-server");
+const neo4j = require("neo4j-driver");
+
+const typeDefs = gql`
+type Post {
+    id: ID! @id
+    content: String!
+    creator: User! @relationship(type: "HAS_POST", direction: IN)
+}
+
+type User {
+    id: ID! @id
+    name: String
+    posts: [Post!]! @relationship(type: "HAS_POST", direction: OUT)
+}
+`;
+
+const driver = neo4j.driver(
+    "bolt://localhost:7687",
+    neo4j.auth.basic("", "")
+);
+
+const neoSchema = new Neo4jGraphQL({
+    typeDefs, driver,
+    config: {
+        driverConfig: {
+            database: "memgraph",
+        },
+    }
+});
+
+neoSchema.getSchema().then((schema) => {
+  const server = new ApolloServer({
+      schema,
+  });
+
+  server.listen().then(({ url }) => {
+      console.log(`🚀 Server ready at ${url}`);
+  });
+})
+```
+
+In this example, it's evident that the data schema must be explicitly defined. The GraphQL server can
+exclusively process queries that operate on data conforming to the specified schemas. In this example
+the schema defines a graph, where the node types *User* and *Post* may be connected by a relationship
+with a type of *HAS_POST*. As it can be seen in this example, to set up a GraphQL client against a running
+Memgraph instance, you have to specify the GraphQL schema, set up the used drivers and specify Memgraph,
+as the used database. The last step is important because the default database name in *driverConfig* is not
+compatible with Memgraph.
```

Comment:
```suggestion
The example above shows that the data schema must be explicitly defined. The GraphQL server can only process queries that operate on data conforming to the specified schemas. In this example, the schema defines a graph, where the nodes labeled as `User` or `Post` may be connected with a `HAS_POST` relationship. 

To set up a GraphQL client against a running Memgraph instance, you have to specify the GraphQL schema, set up the used drivers, and specify Memgraph as the used database. Specifying the used database is especially important because the default database name in the `driverConfig` is not compatible with Memgraph.
```